src/components/Messages/Message.tsx

```diff
@@ -11,14 +14,17 @@ import { useAppStore } from 'src/store/app';
 import { useMessageStore } from 'src/store/message';
 
 const Message: FC = () => {
-  const router = useRouter();
-  const address = router.query.address as string;
-  const messageState = useMessageStore((state) => state);
-  const { messages, conversations } = messageState;
+  const { push, query } = useRouter();
+  const address = query.address as string;
   const currentProfile = useAppStore((state) => state.currentProfile);
+  const messageState = useMessageStore((state) => state);
+  const { conversations } = messageState;
+  const selectedConversation = conversations.get(address);
+  const { messages } = useGetMessages(selectedConversation);
+  const { sendMessage } = useStreamMessages(selectedConversation);
```

Comment:
Not quite sure why `sendMessage` is included in `useStreamMessages`. Feels unrelated to me

Reply:
Also, I noticed that `sendMessage` isn't memoized with `useCallback` when it should be